### PR TITLE
Tweak the heuristic, in `src/core/jpg.js`, that handles JPEG images with a wildly incorrect SOF (Start of Frame) `scanLines` parameter (issue 10989)

### LINKED ISSUE #10989: Single page large format displays as a blank page or incorrectly
This looks to be a CAD type drawing displays as a blank page in FF, Image compressed into top inch of the page in Chrome, and fails in IE 11
Tested in PDF.js Libraries (v2.0.943).

Document to be uploaded to the ticket by customer shortly.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
Fixes #10989

### CODE REVIEW

File: src/core/jpg.js [hunk 1]
```diff
@@ -163,12 +163,16 @@ var JpegImage = (function JpegImageClosure() {
           } else if (nextByte === /* EOI = */ 0xd9) {
             if (parseDNLMarker) {
               // NOTE: only 8-bit JPEG images are supported in this decoder.
-              const maybeScanLines = blockRow * 8;
+              const maybeScanLines = blockRow * (frame.precision === 8 ? 8 : 0);
```
Comment (Snuffleupagus) [inline src/core/jpg.js:166]: Not completely related to the patch, but a better-safe-than-sorry sort of change when I had to touch related code anyway.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Nice find!

/botio makeref

**Pull request merged**